Benchling.js

```diff
@@ -0,0 +1,171 @@
+	if (titleEditable) {
+		shortTitle = titleEditable.textContent.trim();
+	}
+	item.shortTitle = shortTitle;
+	
+	// Find the Display experiment ID from the human-id span
+	var expId = '';
+	var humanId = doc.querySelector('#human-id');
+	if (humanId) {
+		expId = humanId.textContent.trim();
+	}
+	
+	// Fallback: If not found, maybe the metadata tab is active and the ID is in a different location.
+	if (!expId) {
+		var metadataIdSection = doc.querySelector('[data-test-element="humanIdSection"]');
+		if (metadataIdSection) {
+			// Get the text content after the "ID" header
+			var sectionText = metadataIdSection.textContent.trim();
+			// Remove "ID" from the beginning if present
+			expId = sectionText.replace(/^ID\s*/, '').trim();
+		}
+	}
+	
+	// If we found an EXP ID, add it to the title
+	if (expId) {
+		item.title = shortTitle + ' [' + expId + ']';
+	}
+	else {
+		item.title = shortTitle;
+	}
+	
+	// Find authors from metadata field. Only works if the Metadata tab is visible.
+	var authorField = doc.querySelector('.metadata-field--authorList');
+	if (authorField) {
+		var authorLinks = authorField.querySelectorAll('a[href]');
+		for (var i = 0; i < authorLinks.length; i++) {
+			var authorName = authorLinks[i].textContent.trim();
+			if (authorName) {
+				// Try to parse the name into first/last
+				var lastSpaceIndex = authorName.lastIndexOf(' ');
+				if (lastSpaceIndex > 0) {
+					// Has a space, assume "First Last"
+					item.creators.push({
+						firstName: authorName.substring(0, lastSpaceIndex),
+						lastName: authorName.substring(lastSpaceIndex + 1),
+						creatorType: 'author'
+					});
+				}
+				else {
+					// No space, single field
+					item.creators.push({
+						lastName: authorName,
+						creatorType: 'author',
+						fieldMode: 1
+					});
+				}
+			}
+		}
+	}
+	
+	// Find the review status badge. Only works if the Notes tab of the ELN entry is visible.
+	var reviewBadge = doc.querySelector('#badge-button');
+	var extraInfo = [];
+	if (reviewBadge) {
+		var reviewText = reviewBadge.textContent.trim();
+		if (reviewText) {
+			extraInfo.push('Review status: ' + reviewText);
+		}
+	}
+	
+	// Find the project folder. Only works if the Metadata tab is visible.
+	var parentLink = doc.querySelector('[data-test-component="ParentLink"]');
+	if (parentLink) {
+		// Get the text content, which should be the folder name
+		var folderText = parentLink.textContent.trim();
+		// Remove any SVG icons or other elements by getting just the text node
+		var folderName = folderText.split('\n')[0].trim();
+		if (folderName) {
+			item.archiveLocation = 'Project folder: ' + folderName;
+		}
+	}
+	
+	item.libraryCatalog = 'Benchling';
+	item.url = url;
+	
+	if (extraInfo.length > 0) {
+		item.extra = extraInfo.join('\n');
+	}
+	
+	// item.accessDate = '';  // Maps to CSL 'accessed': Date when the document was accessed - Automatically added by Zotero
+	
+	item.complete();
+}
```

Comment:
Use Scaffold to add test cases. You may find it easier in the Zotero 8 beta, which has some improvements to the interface.